build.go

```diff
@@ -313,6 +316,63 @@ func generateSquashfsLayer(oci casext.Engine, name string, author string, opts *
 	return nil
 }
 
+// SaveLayer stores the final layers into a separate location based on the content of
+// the stackerfile, this is useful to avoid an extra manual step to upload build results
+// and also in case of caching in between stacker builds
+// The logic should work for both Docker registry destination and OCI layout destinations
+// In case of OCI layout destinations the tag will be included in the layer name
+func SaveLayer(opts *BuildArgs, sf *Stackerfile, name string) error {
+	if len(sf.buildConfig.SaveUrl) == 0 {
+		fmt.Printf("layer %s will not be save since it doesn't have a save URL\n", name)
+		return nil
+	}
+
+	// Need to determine if URL is docker/oci or something else
+	is, err := NewImageSource(sf.buildConfig.SaveUrl)
+	if err != nil {
+		return err
+	}
+
+	// Determine list of tags to be used
+	tags := opts.RemoteSaveTags
+
+	// Attempt to produce a git commit tag
+	commitTag, err := NewGitLayerTag(sf.referenceDirectory)
+	if err == nil {
+		// Add git tag to the list of tags to be used
+		tags = append(tags, commitTag)
+	}
+
+	if len(tags) == 0 {
+		fmt.Printf("can't save layer %s since list of tags is empty\n", name)
+	}
+
+	// Store the layers to new detination
+	for _, tag := range tags {
+		var destUrl string
+		switch is.Type {
+		case DockerType:
+			destUrl = fmt.Sprintf("%s/%s:%s", strings.TrimRight(sf.buildConfig.SaveUrl, "/"), name, tag)
```

Comment:
I think `path.Join()` would get rid of the need for the `strings.TrimRight()` bit? Not sure if it looks any better or not.

Reply:
Does that work for URLs? I though it's just for file paths on the disk.

Reply:
You're right that it is sort of an abuse of `path.Join()`; it works just fine though, `path.Join()` doesn't actually look at the filesystem at all when doing work. But feel free to ignore this if you feel like it's a hack compared to what you've got now.

Reply:
Not sure path.Join is ok for this, afaik it calls path.Clean which removes double slashes, and may remove the double slashes separating the protocol from the hostname.